guidance/models/_azure_openai.py

```diff
@@ -77,55 +66,61 @@ def __init__(
                 azure_deployment=azure_deployment,
                 tokenizer=tokenizer,
```

Comment:
similar to above, I imagine we may need to make initializing and passing a tokenizer a required arg if we can't autodetect? seems like the default behavior is to pass through None here

Reply:
Users should be able to pass through a particular tokeniser if they want, but otherwise, this should just use the same `model` argument as the OpenAI implementation.

Reply:
Yeah, it's just that the AzureOpenAI model deployments can be named anything, which would likely break the tiktoken detection logic we use in the OpenAI base classes. But that's OK - this is as good as our current behavior, so we can deal with cleaning it up later. It may be worth investigating if there is some way in Azure APIs to detect what the underlying model powering any deployment is (so that we get a tiktoken friendly string).

Reply:
Not if the user has passed in the `model` argument set to the name of the model used by the deployment. We can certainly investigate if there's a way of getting that info, short of querying the Azure API (to which the user might not have access)